### ISSUE TITLE
Questions about the Straight-Through Estimator (STE) operation in counting loss

### ISSUE THREAD

**Issue by JWZhao-uestc** [Melon-Xu/TopoCellGen#3]
Thanks for your awesome work! I’m a bit confused about how the Straight-Through Estimator (STE) is applied in the calculation of the counting loss in the script: **loss_functions/counting_loss.py.** Could you explain how the STE operation is implemented there? 
In addition, using the .item() operation prevents gradients from flowing, but in the code below, .item() is applied in each iteration.

`for i in range(batch_size):

    for j in range(channel_num):

        gt_count = torch.sum(gt_map[i,j,:,:]) / 9.0
   
        pred_count = torch.sum(binarized_pred_map[i,j,:,:]) / 9.0

        channel_loss = torch.abs(gt_count - pred_count)

        losses[f'batch_{i}_channel_{j}'] = channel_loss.item()
        total_loss += channel_loss.item()`

Appreciate it!

**Comment by zihao2951260206-prog** [comment]
请问一下，这个问题您解决了吗？
